lib/colorls/core.rb

```diff
@@ -280,11 +280,21 @@ def chars_for_size
     end
 
     def justify_size_info(size)
-      size_num = size[0][0..-4].rjust(chars_for_size, ' ')
+      size_num = displayable_size_num(size[0]).rjust(chars_for_size, ' ')
```

Comment:
Instead of parsing the string after conversion, just use `filesize.pretty(precision: 0)` and `filesize.to_s('B', precision: 0)` respectively, in the tertiary expression in the line 263 in `size_info`. Then just change this to
```
size_num = size[0].rjust(chars_for_size, ' ')
```